lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -133,7 +133,8 @@ sharedToLinearLayoutNoLeadingOffset(ArrayRef<int64_t> shape,
   int rank = shape.size();
   if (rank == 1) {
     return combineCtaCgaWithShape(
-        LinearLayout::identity1D(shape[0], S("offset"), S("dim0")),
+        LinearLayout::identity1D(shape[0] / shared.getCTASplitNum()[0],
```

Comment:
Move the call to `getShapePerCTA` above this if and use `shapePerCTA` here. Similar suggestion applies to all the changes in this file.